api/team_token.go

```diff
@@ -38,6 +38,34 @@ func tokenList(w http.ResponseWriter, r *http.Request, t auth.Token) error {
 	return json.NewEncoder(w).Encode(tokens)
 }
 
+// title: token info
+// path: /tokens/{token_id}
+// method: GET
+// produce: application/json
+// responses:
+//   200: List token
+//   204: No content
+//   401: Unauthorized
+func tokenInfo(w http.ResponseWriter, r *http.Request, t auth.Token) error {
+	tokenID := r.URL.Query().Get(":token_id")
+	if tokenID == "" {
+		w.WriteHeader(http.StatusBadRequest)
+		return nil
+	}
+	token, err := servicemanager.TeamToken.FindByTokenID(tokenID)
+	if err != nil {
+		if err == authTypes.ErrTeamTokenNotFound {
+			return &errors.HTTP{
+				Code:    http.StatusNotFound,
+				Message: err.Error(),
+			}
+		}
+		return err
+	}
+	w.Header().Set("Content-Type", "application/json")
```

Comment:
I'm sorry I forgot about this part when we were discussing this issue.
We need to check for permission to read the found token, for team tokens we have two different checks that need to be made, the first one is just like the other handlers checking if we can read the token:

```
allowed := permission.Check(t, permission.PermTeamTokenRead,
    permission.Context(permTypes.CtxTeam, teamToken.Team),
)
if !allowed {
    return permission.ErrUnauthorized
}
```
The second is particular to team tokens, we only let the user see the token value if the token permissions is a subset of user permissions, if it's not we must set the token value to an empty string. The logic will be the same as in https://github.com/tsuru/tsuru/blob/a80bbcacaec00b4e268cfe1cc023456427c4bed3/auth/team_token.go#L229-L235

I think this logic can be extracted to a new public function and this handler will call it.